aviary/subsystems/propulsion/engine_scaling.py

```diff
@@ -130,107 +144,86 @@ def compute(self, inputs, outputs):
 
             scale_factor = engine_scale_factor
 
-        outputs[Dynamic.Mission.THRUST] = unscaled_net_thrust * scale_factor
-        outputs[Dynamic.Mission.THRUST_MAX] = unscaled_max_thrust * scale_factor
-        # user-specified constant_fuel_flow value is currently not scaled with engine
-        outputs[Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE] = - \
-            (unscaled_fuel_flow_rate * fuel_flow_scale_factor) - constant_fuel_flow
-
-        # all other variables are just linearly scaled
-        outputs[Dynamic.Mission.ELECTRIC_POWER_IN] = unscaled_electric_power_in * scale_factor
-        outputs[Dynamic.Mission.NOX_RATE] = unscaled_nox_rate * scale_factor
-        outputs[Dynamic.Mission.SHAFT_POWER] = unscaled_shaft_power * scale_factor
-        outputs[Dynamic.Mission.SHAFT_POWER_MAX] = unscaled_shaft_power_max * scale_factor
-
-        # outputs[Dynamic.Mission.EXIT_AREA] = unscaled_exit_area * scale_factor
+        # loop through all variables, singling out fuel flow to have special scaling
+        # compute 'max' counterpart of variables that have them
+        for variable in engine_variables:
+            if variable not in skip_variables:
+                if variable is FUEL_FLOW:
+                    outputs[Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE] = (
+                        -(inputs['fuel_flow_rate_unscaled'] * fuel_flow_scale_factor)
+                        - constant_fuel_flow
```

Comment:
I think
`-(inputs['fuel_flow_rate_unscaled'] * fuel_flow_scale_factor + constant_fuel_flow)`
is a little clearer